oioswift/common/middleware/container_hierarchy.py

```diff
@@ -192,8 +193,11 @@ def hset(self, key, path, val):
     def hsetnx(self, key, path, val):
         return self.conn.hsetnx(key, path, val)
 
-    def zsetnx(self, key, path):
-        return self.conn.zadd(key, {path: 1}, nx=True)
+    def zset(self, key, path):
```

Comment:
Maybe we should compute once LooseVersion in `__init__` method
or having `zset_legacy` `zset3` and in `__init__`:
```
if  if LooseVersion(self.__redis_mod.__version__) < LooseVersion("3.0.0"):
    self.zset = self.zset_legacy
else:
    self.zset = self.zset3
```